tools/lsp/common/document_cache.rs

```diff
@@ -333,7 +334,11 @@ impl DocumentCache {
     }
 
     pub fn drop_document(&mut self, url: &Url) -> Result<()> {
```

Comment:
```suggestion
    pub fn drop_document(&mut self, url: &Url){
```

The function shouldn't probably return an error if it cannot be an error.

Reply:
It seems that drop_document on the type loader can still produce an error.